docs/pages.jl

```diff
@@ -9,7 +9,7 @@ pages = Any[
         "model_creation/dsl_description.md",
         # DSL - Advanced
         "model_creation/programmatic_CRN_construction.md",
-        "model_creation/constraint_equations.md",
+        "model_creation/coupled_crn_equations.md",
```

Comment:
The old file is still around. Do you think it would make sense to turn it into an example of how to combine equations + events + CRN models? It is a bit more straightforward with the examples, while this new one simply goes through all options one by one.